File: src/com/basmilius/ps/bastools/component/BTCodeStyleScheme.java
package com.basmilius.ps.bastools.component;

import com.intellij.psi.codeStyle.CodeStyleScheme;
import com.intellij.psi.impl.source.codeStyle.CodeStyleSchemeImpl;
import org.jetbrains.annotations.NotNull;

public abstract class BTCodeStyleScheme extends CodeStyleSchemeImpl
{

	private final CodeStyleScheme parentScheme;

	public BTCodeStyleScheme (@NotNull String name, boolean isDefault, CodeStyleScheme parentScheme)
	{
		super(name, isDefault, parentScheme);

		this.parentScheme = parentScheme;
	}

	public final <T extends CodeStyleSchemeImpl> T getParentScheme (Class<T> implementation)
	{
		return implementation.cast(this.parentScheme);
	}
}

File: src/com/basmilius/ps/bastools/component/BTCodeStyleScheme.kt
package com.basmilius.ps.bastools.component

import com.intellij.psi.codeStyle.CodeStyleScheme
import com.intellij.psi.impl.source.codeStyle.CodeStyleSchemeImpl

abstract class BTCodeStyleScheme(name: String, isDefault: Boolean, private val parentScheme: CodeStyleScheme?) : CodeStyleSchemeImpl(name, isDefault, parentScheme)
{

	/**
	 * Gets the parent scheme.
	 */
	fun <T : CodeStyleSchemeImpl> getParentScheme(implementation: Class<T>): T
	{
		return implementation.cast(this.parentScheme)
	}

}
